src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubListener.cs

```diff
@@ -38,77 +38,115 @@ void IListener.Cancel()
             // nothing to do. 
         }
 
+        // This will get called once when starting the JobHost. 
         public Task StartAsync(CancellationToken cancellationToken)
-        {            
+        {
             return _eventListener.RegisterEventProcessorFactoryAsync(this, _options);
         }
 
         public Task StopAsync(CancellationToken cancellationToken)
         {
             return _eventListener.UnregisterEventProcessorAsync();
         }
-
-        Task IEventProcessor.OpenAsync(PartitionContext context)
+        
+        // This will get called per-partition. 
+        IEventProcessor IEventProcessorFactory.CreateEventProcessor(PartitionContext context)
         {
-            // Begin listener 
-            return Task.FromResult(0);
+            return new Listenter(this);
         }
 
-        async Task IEventProcessor.ProcessEventsAsync(PartitionContext context, IEnumerable<EventData> messages)
+        // We get a new instance each time Start() is called. 
+        // We'll get a listener per partition - so they can potentialy run in parallel even on a single machine.
+        private class Listenter : IEventProcessor
         {
-            EventHubTriggerInput value = new EventHubTriggerInput
+            private readonly EventHubListener _parent;
+            private readonly CancellationTokenSource _cts = new CancellationTokenSource();
+            private static Task _done = Task.FromResult(0);
```

Comment:
Where exactly is that defined?  I don't see a completed property on System.Threading.Tasks.Task and I don't see any other reference to that in the SDK.